File: src/desktop/lib/models/announcement/update_announcement_request.dart
class UpdateAnnouncementRequest {
  final String title;
  final String content;
  final bool? isDeleted;

  UpdateAnnouncementRequest({
    required this.title,
    required this.content,
    this.isDeleted,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{'title': title, 'content': content};

    if (isDeleted != null) {
      json['isDeleted'] = isDeleted;
    }

    return json;
  }

  factory UpdateAnnouncementRequest.fromJson(Map<String, dynamic> json) {
    return UpdateAnnouncementRequest(
      title: json['title'] as String,
      content: json['content'] as String,
      isDeleted: json['isDeleted'] as bool?,
    );
  }

  @override
  String toString() {
    return 'UpdateAnnouncementRequest(title: $title, content: $content, isDeleted: $isDeleted)';
  }
}

File: src/api/MoRS.ManagementSystem.Application/DTOs/Announcement/UpdateAnnouncementRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MoRS.ManagementSystem.Application.DTOs.Announcement;

public class UpdateAnnouncementRequest
{
    [Required]
    [StringLength(100, MinimumLength = 5)]
    public required string Title { get; set; }

    [Required]
    [StringLength(2000, MinimumLength = 10)]
    public required string Content { get; set; }
    public bool IsDeleted { get; set; } = false;
}
